docs/content/policy-performance.md

```diff
@@ -333,6 +333,190 @@ Another way to get the same output as above would be the following:
+| histogram_timer_rego_query_eval_ns_99.9%  |     335906 |
+| histogram_timer_rego_query_eval_ns_99.99% |     336493 |
+| histogram_timer_rego_query_eval_ns_mean   |      40355 |
+| histogram_timer_rego_query_eval_ns_median |      35846 |
+| histogram_timer_rego_query_eval_ns_99%    |     133936 |
+| histogram_timer_rego_query_eval_ns_90%    |      44780 |
+| histogram_timer_rego_query_eval_ns_95%    |      50815 |
+| histogram_timer_rego_query_eval_ns_min    |      31284 |
+| histogram_timer_rego_query_eval_ns_max    |     336493 |
+| histogram_timer_rego_query_eval_ns_75%    |      38254 |
+| histogram_timer_rego_query_eval_ns_count  |      37396 |
++-------------------------------------------+------------+
+```
+
+These results represent basically the _best_ times than the evaluation can be done on the local machine. All time spent
+preparing for the evaluation (parsing, compiling, etc) is omitted.
+
+#### Options for `opa bench`
+| Option | Detail | Default |
+| --- | --- | --- |
+| <span class="opa-keep-it-together">`--benchmem`</span> | Report memory allocations with benchmark results. | true |
+| <span class="opa-keep-it-together">`--metrics`</span> | Report additional query performance metrics. | true |
+| <span class="opa-keep-it-together">`--count`</span> | Number of times to repeat the benchmark. | 1 |
+
+
+### Benchmarking OPA Tests
+
+There is also a `--bench` option for `opa test` which will perform benchmarking on OPA unit tests. This will evaluate
+any loaded tests as benchmarks. There will be additional time for any test-specific actions are included so the timing
+will typically be longer than what is seen with `opa bench`. The primary use-case is not for absolute time, but to
+track relative time as policies change.
+
+#### Options for `opa test --bench`
+| Option | Detail | Default |
+| --- | --- | --- |
+| <span class="opa-keep-it-together">`--benchmem`</span> | Report memory allocations with benchmark results. | true |
+| <span class="opa-keep-it-together">`--count`</span> | Number of times to repeat the benchmark. | 1 |
+
+
+#### Example Tests
+Adding a unit test file for the [policy source as shown above](#example-policy):
+
+```rego
+package rbac
+
+
+test_user_has_role_dev {
+    user_has_role["dev"] with input as {"subject": "alice"}
+}
+
+test_user_has_role_negative {
+    not user_has_role["super-admin"] with input as {"subject": "alice"}
+}
+```
+
+Which when run normally will output something like:
+```
+$ opa test -v ./rbac.rego ./rbac_test.rego
+data.rbac.test_user_has_role_dev: PASS (605.076µs)
+data.rbac.test_user_has_role_negative: PASS (318.047µs)
+--------------------------------------------------------------------------------
+PASS: 2/2
+```
+
+#### Example: Benchmark rbac unit tests
+
+```bash
+opa test -v --bench ./rbac.rego ./rbac_test.rego
+```
+Results in output:
+```
+data.rbac.test_user_has_role_dev	   44749	     27677 ns/op	     23146 timer_rego_query_eval_ns/op	   12303 B/op	     229 allocs/op
```

Comment:
🤔 I'm a little confused here -- when you do `opa bench`, it'll return the histograms, but when you do `opa test --bench`, it's `/op`.

Reply:
Oops, I didn't mean to include this comment. It's fine 👍

Reply:
Yea, to clarify, `opa test --bench` gives the average for the eval since there is likely additional testing overhead the actual exact numbers are not really as useful. The main use-case for this is catching changes in policy testing times, you want to see relative changes to the average. Then `opa bench` is more for the like the "give me exactly how long this query takes" use-case. Over there we include a significant amount of additional data.

In theory there isn't any reason we can't do the histogram for test too, just didn't seem as important.

Reply:
That makes sense, thanks. :)